src/unitxt/stream.py

```diff
@@ -89,6 +93,41 @@ def take(self, n):
             yield instance
 
 
+class DynamicStream(Stream):
+    generator: Callable
+    gen_kwargs: Dict[str, Any] = OptionalField(default_factory=dict)
+    caching: bool = False
+    copying: bool = False
+
+    def __post_init__(self):
+        self.stream = None
+        if settings.use_eager_execution:
+            try:
+                instances_list = []
+                for instance in self.generator(**self.gen_kwargs):
+                    instances_list.append(instance)
+                self.stream = ListStream(instances_list=instances_list)
+            except:
+                # input stream is empty
```

Comment:
This should except only that exception (If needed we can add EmptyStreamException)

Reply:
A very good point, @elronbandel . Will dig into this.